File: #21_Merge_Two_Sorted_Lists/Solution.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    # @param {ListNode} l1
    # @param {ListNode} l2
    # @return {ListNode}
    def mergeTwoLists(self, l1, l2):
        if not l1:
        	return l2
        if not l2:
        	return l1
        dummy = ListNode(-1)
        cur = dummy
        while l1 or l2:
        	if not l1:
        		node = l2
        	elif not l2:
        		node = l1
        	else:
        		if l1.val < l2.val:
        			node = l1
        		else:
        			node = l2
        	cur.next = node
        	if node == l1:
        		l1 = l1.next
        	else:
        		l2 = l2.next
        	cur = cur.next
        return dummy.next

File: #21_Merge_Two_Sorted_Lists/Solution.java
class ListNode {
    int val;
    ListNode next;
    ListNode(int x) { val = x; }
}

public class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        if (l1 == null) return l2;
        if (l2 == null) return l1;
        ListNode cur1 = l1;
        ListNode cur2 = l2;
        ListNode newCur = null;
        ListNode newHead = null;
        while (cur1 != null && cur2 != null) {
        	ListNode node = cur2;
        	if (cur1.val < cur2.val) node = cur1;
    		if (newHead == null) {
    			newHead = node;
        		newCur = newHead;
    		}
    		else {
	    		newCur.next = node;
	    		newCur = node;
    		}
    		if (node == cur2) cur2 = cur2.next;
    		else cur1 = cur1.next;
        }
        while (cur1 != null) {
        	newCur.next = cur1;
        	newCur = cur1;
        	cur1 = cur1.next;
        }
        while (cur2 != null) {
        	newCur.next = cur2;
        	newCur = cur2;
        	cur2 = cur2.next;
        }
        return newHead;
    }
}